tests/integration/test_integration.py

```diff
@@ -0,0 +1,221 @@
+                "juju_application": f"{PROMETHEUS_CONFIGURER_APP_NAME}",
+                "juju_charm": f"{PROMETHEUS_CONFIGURER_APP_NAME}",
+                "juju_model": f"{model_name}",
+                "juju_model_uuid": f"{model_uuid}",
+                "networkID": f"{TEST_TENANT}",
+                "severity": "Low",
+            },
+            "lastEvaluation": "0001-01-01T00:00:00Z",
+            "name": f"{TEST_ALERT_NAME}",
+            "query": f'process_cpu_seconds_total{{juju_application="{PROMETHEUS_CONFIGURER_APP_NAME}",juju_charm="{PROMETHEUS_CONFIGURER_APP_NAME}",juju_model="{model_name}",juju_model_uuid="{model_uuid}",networkID="{TEST_TENANT}"}} '  # noqa: E501
+            "> 0.12",
+            "state": "inactive",
+            "type": "alerting",
+        }
+
+        server_response = requests.post(
+            f"http://{prometheus_configurer_server_ip}:9100/{TEST_TENANT}/alert",
+            json=test_rule_json,
+        )
+        assert server_response.status_code == 200
+
+        # Prometheus needs a couple of seconds to reload configuration
+        time.sleep(5)
+        prometheus_rules = await _get_prometheus_rules(ops_test, PROMETHEUS_APP_NAME, 0)
+
+        assert len(prometheus_rules) == 1
+        assert prometheus_rules[0]["rules"][0] == expected_prometheus_rules
+
+    @pytest.mark.abort_on_fail
+    async def test_given_prometheus_configurer_with_existing_rule_when_get_alert_rules_then_expected_alert_rules_are_returned(  # noqa: E501
+        self, ops_test: OpsTest, setup
+    ):
+        prometheus_configurer_server_ip = await _unit_address(
+            ops_test, PROMETHEUS_CONFIGURER_APP_NAME, 0
+        )
+        expected_response = {
+            "alert": f"{TEST_ALERT_NAME}",
+            "expr": f'process_cpu_seconds_total{{networkID="{TEST_TENANT}"}} > 0.12',
+            "for": "0s",
+            "labels": {"networkID": f"{TEST_TENANT}", "severity": "Low"},
+            "annotations": {"description": "Rule description.", "summary": "Rule summary."},
+        }
+
+        server_response = requests.get(
+            f"http://{prometheus_configurer_server_ip}:9100/{TEST_TENANT}/alert"
+        )
+        assert server_response.status_code == 200
+
+        assert server_response.json()[0] == expected_response
+
+    @pytest.mark.abort_on_fail
+    async def test_given_prometheus_configurer_with_existing_rule_when_delete_alert_rule_then_prometheus_configurer_has_no_rules(  # noqa: E501
+        self, ops_test: OpsTest, setup
+    ):
+        prometheus_configurer_server_ip = await _unit_address(
+            ops_test, PROMETHEUS_CONFIGURER_APP_NAME, 0
+        )
+        server_response = requests.delete(
+            f"http://{prometheus_configurer_server_ip}:9100/{TEST_TENANT}/alert?alert_name={TEST_ALERT_NAME}"  # noqa: E501, W505
+        )
+        assert server_response.status_code == 204
+        # 2 seconds for the Prometheus Configurer to do what needs to process the request
```

Comment:
2 or 5?

Reply:
5 :( GH workers are too slow to work with 2 sec :/